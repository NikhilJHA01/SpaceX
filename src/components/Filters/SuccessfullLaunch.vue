<template>
  <el-checkbox-group
    v-model="launchSucessArray"
    size="small"
    text-color="#000000"
    fill="#5daf34"
    @change="changeFilter()"
  >
    <el-checkbox-button v-for="value in options" :label="value" :key="value">{{
      value
    }}</el-checkbox-button>
  </el-checkbox-group>
</template>

<script>
import { BooleanArray } from "../../utility/constant.js";

export default {
  name: "SuccessfullLaunch",
  data() {
    return {
      launchSucessArray: [],
      options: BooleanArray
    };
  },
  methods: {
    changeFilter() {
      this.$store.dispatch(
        "Set_Filter_By_Launch_Success",
        this.launchSucessArray
      );
      this.$emit("changeFilter");
    }
  }
};
</script>
