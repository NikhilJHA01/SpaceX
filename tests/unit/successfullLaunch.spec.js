import { shallowMount } from "@vue/test-utils";
import SuccessfullLaunch from "@/components/Filters/SuccessfullLaunch";

describe("SuccessfullLaunch Component", () => {
  const wrapper = shallowMount(SuccessfullLaunch);

  it("renders the SuccessfullLaunch", async () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has a checkbox-group", () => {
    const form = wrapper.find("el-checkbox-group");
    expect(form.exists()).toBe(true);
  });

  it("Filter is not selected by default", () => {
    expect(wrapper.vm.$data.launchSucessArray).toHaveLength(0);
  });
});
