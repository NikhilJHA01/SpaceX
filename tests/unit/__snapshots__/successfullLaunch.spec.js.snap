// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessfullLaunch Component renders the SuccessfullLaunch 1`] = `
<el-checkbox-group size="small" text-color="#000000" fill="#5daf34">
  <el-checkbox-button label="TRUE">TRUE</el-checkbox-button>
  <el-checkbox-button label="FALSE">FALSE</el-checkbox-button>
</el-checkbox-group>
`;
